Remove read300 functionality

dmitrio95 opened the pull request:
This pull request contains changes related to removal of support for reading "old" 3.0 MSCX fileformat, as it was discussed in #3760 and #3870. Most of `read300` functions that provided this functionality were simply removed but some of `read300` functions that largely differ from `read` functions in their current state were moved to `read206.cpp`.

In order to avoid possible errors I have added a check which raises an error if the currently read file was written in "old" 3.0 MSCX format (see d2d1386d68bd8c7fee6b71a00599d0213b2a5b9f). Probably some proper message with a link to some build which is able to read this format should be added for that case but I have not done it for now.

anatoly-os commented:
Thank you @dmitrio95. I and @wschweer will have a look within few days.

anatoly-os commented:
@dmitrio95 could you rebase your PR please? I played with your changes and everything seems ok.

dmitrio95 commented:
@anatoly-os I rebased this pull request. One test has failed but in this case test seems to be wrong: `<play>0</play>` properties in some fermatas were missing in the reference MSCX file though they were originally present in the test 2.X file. I have added a05228fb901a620e48c20f4df1a8783483a3e938, perhaps my changes have somehow unintentionally fixed another versions incompatibility :)
